test/integration/connection-test.coffee

```diff
@@ -1020,3 +1020,55 @@ exports.disableAnsiNullDefault = (test) ->
     runSqlBatch test, config, sql, (err) ->
         test.ok(err instanceof Error)
         test.strictEqual err?.number, 515 # Cannot insert the value NULL
+
+# Test that the default setting for DATEFIRST is 7
+exports.testDatefirstDefault = (test) ->
+  test.expect(3)
+
+  config = getConfig()
+
+  connection = new Connection(config)
+
+  request = new Request('select @@datefirst', (err) ->
+    test.ifError(err)
+    connection.close()
+  )
+
+  request.on('row', (columns) ->
+    dateFirst = columns[0].value
+    test.strictEqual(dateFirst, 7)
+  )
+
+  connection.on 'connect', (err) ->
+    test.ifError(err)
+    connection.execSql(request)
+
+  connection.on 'end', (info) ->
+    test.done()
+
+# Test that the DATEFIRST setting can be changed via an optional configuration
+exports.testDatefirstDefault = (test) ->
```

Comment:
Since these tests are identical, these can be refactored as below:

```JavaScript
function testDateFirstImpl(test, dateFirstSetting)
..... Implementation here. ....

exports.testDateFirstDefault = (test) ->
    testDateFirstImpl(test, undefined)

exports.testDateFirstNonDefault = (test) ->
    testDateFirstImpl(test, 3)
```